### PR TITLE
Move components from prom stack to alloy helm

### PR DESCRIPTION
**Pull request by samidbb**
## Describe your changes
This PR move deployments of Prom stack components to Grafana agent Helm chart
Note: 
- When deploying resources using this release, open service window to notify about a possible outage in metrics collection. 
- When deploying resources using this release, set "grafana_agent_namespace" variable to "grafana"

## Issue ticket number and link
https://github.com/dfds/cloudplatform/issues/2804

## Checklist before requesting a review
- [x] I have tested changes in my sandbox
- [ ] I have added the needed changes in the `test/integration` folder to apply my changes in QA. [Read the guide on adding environment variables in QA](https://wiki.dfds.cloud/en/ce-private/atlantis/adding-env-vars)
- [x] I have rebased the code to master (or merged in the latest from master)

## Is it a new release?
- [x] Apply a release tag `release:(major|minor|patch)`, following semantic versioning in [this guide](https://semver.org/) or `norelease` if there is no changes to the Terraform code

### CODE REVIEW

File: _sub/compute/helm-kube-prometheus-stack/values/node-exporter.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
-prometheus-node-exporter:
-  resources:
-    requests:
-      cpu: 20m
-      memory: 50Mi
-  priorityClassName: ${prometheus_node_exporter_priorityclass}
-  prometheus:
-    monitor:
-      enabled: true
+# prometheus-node-exporter:
```
Comment (avnes) [inline _sub/compute/helm-kube-prometheus-stack/values/node-exporter.yaml:1]: I guess these comments are temporarily and will be removed in a later PR?

File: _sub/monitoring/helm-grafana-agent/main.tf [hunk 2]
```diff
@@ -10,25 +10,28 @@ resource "helm_release" "grafana_agent" {
...
+      enable_side_by_side                    = var.enable_side_by_side,
+      agent_resource_memory_limit            = var.agent_resource_memory_limit,
+      agent_resource_memory_request          = var.agent_resource_memory_request,
+      tolerations                            = var.tolerations,
+      affinity                               = var.affinity,
+      agent_replicas                         = var.agent_replicas,
+      storage_enabled                        = var.storage_enabled,
+      storage_class                          = var.storage_class,
+      storage_size                           = var.storage_size,
+      prometheus_node_exporter_priorityclass = var.priority_class,
```
Comment (avnes) [inline _sub/monitoring/helm-grafana-agent/main.tf:32]: Is it on purpose that both  prometheus_node_exporter_priorityclass and  kube_state_metrics_priorityclass are using the same var.priority_class?

File: _sub/compute/helm-kube-prometheus-stack/values/node-exporter.yaml (hunk 1 above)
Comment (samidbb) [reply _sub/compute/helm-kube-prometheus-stack/values/node-exporter.yaml:1]: I am keeping them here for reference, and they will removed when we remove this module

File: _sub/monitoring/helm-grafana-agent/main.tf (hunk 2 above)
Comment (samidbb) [reply _sub/monitoring/helm-grafana-agent/main.tf:32]: well this is how it is in the current kube-prom-stack if you look under _sub/compute/helm-kube-prometheus-stack/main.tf. I don't have the background info for our choice on priority classes

**Review by Mik-Nord — APPROVED** [review]

**Review by rifisdfds — APPROVED** [review]

**Review by avnes — APPROVED** [review]

**Pull request merged**